From 76ca60f66e3ebe33a47889462b306e9c3d9bdad9 Mon Sep 17 00:00:00 2001
From: Chia-Ping Tsai <chia7712@gmail.com>
Date: Fri, 1 Dec 2017 04:01:15 +0800
Subject: [PATCH 458/468] HBASE-19350 TestMetaWithReplicas is flaky

(cherry picked from commit 9ab637c2292e36313b22b41be1e86809c6ac60f6)

Change-Id: I26e0d6f7eb8399a9c192f2387cc40cb23cb1eb49
Author: Chia-Ping Tsai
Reason: Bug
Ref: CDH-66315
---
 .../org/apache/hadoop/hbase/ClusterStatus.java     |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/hbase-client/src/main/java/org/apache/hadoop/hbase/ClusterStatus.java b/hbase-client/src/main/java/org/apache/hadoop/hbase/ClusterStatus.java
index c8caa96..9f00373 100644
--- a/hbase-client/src/main/java/org/apache/hadoop/hbase/ClusterStatus.java
+++ b/hbase-client/src/main/java/org/apache/hadoop/hbase/ClusterStatus.java
@@ -276,7 +276,10 @@ public class ClusterStatus extends VersionedWritable {
 
   @InterfaceAudience.Private
   public Map<String, RegionState> getRegionsInTransition() {
-    return this.intransition;
+    if (intransition == null) {
+      return Collections.EMPTY_MAP;
+    }
+    return Collections.unmodifiableMap(intransition);
   }
 
   public String getClusterId() {
-- 
1.7.9.5

